### PR TITLE
fix-transferRole-check-on-esdtTransfer

### PR DESCRIPTION
**Pull request by sasurobert**
integrated new flag

### CODE REVIEW

**Review by andrei-marinica — DISMISSED** [review]

**Review by dragos-rebegea — APPROVED** [review]

File: genesis/process/shardGenesisBlockCreator.go
Comment (iulianpascalau) [inline genesis/process/shardGenesisBlockCreator.go:364]: this was a forgotten flag. Does the code is still backwards compatible now?

File: go.mod
Comment (iulianpascalau) [inline go.mod:15]: proper release

**Review by popenta — APPROVED** [review]
@@                            Log scanner                                  @@
##                            fix-transferRole-check-on-transfer           ##
================================================================================
- Known Warnings              450
- New Warnings                342
- Known Errors                0
- New Errors                  0
- Panics                      0
================================================================================

**Review by iulianpascalau — CHANGES_REQUESTED** [review]
Looks good :+1: 
We will wait until the `feat/release-2022-April` gets merged and then re-target this in `rc/2022-may` 
Requested changes only to prevent accidental merges

File: genesis/process/shardGenesisBlockCreator.go
Comment (iulianpascalau) [reply genesis/process/shardGenesisBlockCreator.go:364]: having a second thought, let's keep it this way as this is the correct way. Will deal with the import-db process afterwards

**Review by iulianpascalau — APPROVED** [review]

**Review by andrei-marinica — APPROVED** [review]

**Review by LucianMincu — APPROVED** [review]
System tests passed.

**Pull request merged**
